module/Swissbib/src/Swissbib/Controller/DetailPageController.php

```diff
@@ -147,4 +148,60 @@ protected function getResultsManager()
     {
         return $this->serviceLocator->get('VuFind\SearchResultsPluginManager');
     }
+
+    /**
+     * Gets the Tagcloud
+     *
+     * @param array $subjectIds All subject ids, including duplicates
+     * @param array $subjects   All subjects
+     *
+     * @return array
+     */
+    protected function getTagCloud(array $subjectIds, array $subjects)
+    {
+        $counts = array_count_values($subjectIds);
+        $cloud = [];
+        $max = max($counts);
+
+        foreach ($counts as $id => $count) {
+            $filtered = array_filter(
+                $subjects,
+                function (ESSubject $item) use ($id) {
+                    return $item->getFullUniqueID() === $id;
+                }
+            );
+            if (count($filtered) > 0) {
+                // @var ESSubject $subject
+                $subject = array_shift($filtered);
+
+                $subject->setAbsoluteFrequency($count);
+                $subject->setRelativeFrequency(
+                    $this->calculateFontSize($count, $max)
+                );
+
+                $cloud[] = $subject;
+
+            }
+        }
+
+        return $cloud;
+    }
+
```

Comment:
Changed the behavior of font-size calculation to allow for more precise rendering and also for relative size units like em or rem

Reply:
Ok, but should be moved to Swissbib\Controller\Plugin\TagCloud